src/int/sign.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{ConstChoice, ConstCtOption, Int, Uint};
+
+impl<const LIMBS: usize> Int<LIMBS> {
+    /// Construct new [`Int`] from a sign and magnitude.
+    /// Returns `None` when the magnitude does not fit in an [`Int<LIMBS>`].
+    pub const fn new_from_abs_sign(
+        abs: Uint<LIMBS>,
+        is_negative: ConstChoice,
+    ) -> ConstCtOption<Self> {
+        let magnitude = Self(abs).wrapping_neg_if(is_negative);
+        let fits = Uint::gt(&abs, &Int::MAX.0)
+            .not()
+            .or(is_negative.and(Uint::eq(&abs, &Int::MIN.0)));
+        ConstCtOption::new(magnitude, fits)
+    }
+
+    /// Whether this [`Int`] is negative, as a `ConstChoice`.
+    pub const fn is_negative(&self) -> ConstChoice {
+        ConstChoice::from_word_msb(self.0.to_words()[LIMBS - 1])
+    }
+
+    /// Whether this [`Int`] is positive, as a `ConstChoice`.
+    pub const fn is_positive(&self) -> ConstChoice {
+        Int::is_negative(self).not().and(Int::is_nonzero(self))
+    }
+
+    /// The sign and magnitude of this [`Int`].
+    pub const fn abs_sign(&self) -> (Uint<LIMBS>, ConstChoice) {
+        let sign = self.is_negative();
+        // Note: this negate_if is safe to use, since we are negating based on self.is_negative()
+        let abs = self.wrapping_neg_if(sign);
+        (abs.0, sign)
+    }
```

Comment:
What I'd propose:
- stick with `sign_magnitude`
- implement `abs` to align with `num_traits::Signed`

Reply:
@erik-3milabs 

> I have a couple questions regarding this choice:
> Although technically correct, I think that it is best to use `mag`(nitude) instead of `abs`(olute) for the following reasons:
> 1. when named `sgn_mag`, the function name conveys that it is (also) the representation conversion from U2 to sign-and-magnitude.

The user should not care what is the representation and what conversion was done, what user cares about is that the method returns absolute value (and sign optionally) regardless of what representation is.

> 2. sticking with the representation name: it is sign-and-magnitude, not sign-and-absolute.

But we (I mean mostly you :)) did't implemented it as sign-and-magnitude representation.
See my first comment. User don't care about the representation. It knows it has signed integer and want to have absolute value of that (signed) integer. I am not against using `mag`(nitude) internally in the implementation but user facing interface IMHO should use `abs` as this is very common in other libraries (and did I mention I like consistency?).
All of these common big int implementations use `abs` (not `magnitude`), hence my strong opinion is we should use `abs` as well:

https://pkg.go.dev/math/big#Int.Abs
https://docs.rs/num-bigint/latest/num_bigint/struct.BigInt.html#method.abs
https://gmplib.org/manual/Integer-Arithmetic#index-mpz_005fabs
https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/math/BigInteger.html#abs()
https://learn.microsoft.com/en-us/dotnet/api/system.numerics.biginteger.abs?view=net-8.0#system-numerics-biginteger-abs(system-numerics-biginteger)

> 3. [`num_bigint::BigInt`](https://docs.rs/num-bigint/latest/num_bigint/struct.BigInt.html#method.magnitude) (the other rust library supporting bigint) also uses `magnitude` instead of `absolute`

This crate uses `abs` as well (see above), the `magnitude` method is there because this is how it's implemented and `magnitude` method returns part of the internal state rather than absolute value (which in this case it happens to be the same), but my general comment hold, most opinionated big int implementation use `abs`, `magnitude` is the method to return part of the underlaying state if they happen to implement signed arithmetic as magnitude-sign instead of u2.

> What is your reason for proposing to use `abs`(olute)?

consistency :)
see above other big int implementations

> Note: even if we revert to using `magnitude` I would be OK with supporting `fn abs()`; at that point it refers to the mathematical function that maps `x` to `|x|`. I believe there is even a trait for it that we can implement.

yes, this one: https://docs.rs/num-traits/latest/num_traits/sign/trait.Signed.html 
I am ok using `magnitude` internally

> Can you elaborate on why you chose `abs_sgn` instead of `sgn_abs`? To me, `sgn_abs` feels more natural, but that is because in the representation name "sign-and-magnitude", the sign comes first.

Again this is for consistency.
I prefer consistency with what is already in this create (i.e. least significant first) rather than what is the name of the representation we chose not to use anyway :)
Every method on the Uint when returns a tuple, the first element is always the least significant (there was even PR to change split/mul_split to change it to return (L, H) instead of (H, L).
In our case, The sign is "kind of" most significant (e.g. in 4-bit integers the bit values are `(1, 2, 4, -8)` in that case the sign is determined by the MSB, hence the sign should be last in the tuple.

> Even though I love abbreviations, I still think one should use them sparingly. Could you elaborate on why you feel `abs_sgn` is better than `absolute_sign`, `sign_absolute`, or `sign_magnitude`? (I really like that last one 🙈

Again, for consistency with https://docs.rs/num-traits/latest/num_traits/sign/trait.Signed.html which abbreviates to `abs` and all other libraries I referenced above.

Reply:
Thank you for this very elaborate explanation! You have me convinced. Let's stick with `abs_sign` :+1:

Reply:
One remaining remark: now that we are moving away from `sign_and_magnitude`, could you please update the docstring of this function to reflect this?

Reply:
of course 👍  i'll update the PR in the late evening CEST.